modules/light-clients/06-solomachine/types/update_test.go

```diff
@@ -180,3 +180,386 @@ func (suite *SoloMachineTestSuite) TestCheckHeaderAndUpdateState() {
 		}
 	}
 }
+
+func (suite *SoloMachineTestSuite) TestVerifyClientMessageHeader() {
+	var (
+		clientMsg   exported.Header // TODO: Update to ClientMessage interface
```

Comment:
will we combine the test cases later?

Reply:
Sure, I added two test functions here initially to separate things out a bit but I'd be for combining them later. Once the storage updates are added to `UpdateState` and `UpdateStateOnMisbehaviour` and `CheckerHeaderAndUpdateState` is removed I can add separate tests for those scenarios.